### PR TITLE
Connect mode enable listen so that current node is connectable too

### LINKED ISSUE #11: Node connection not working for multiple nodes
I am hitting this weird issue where if I spawn 3 nodes and try to make the following connection between them

Node1 <--- Node2 <--- Node3 (<--- : "connects to")

I get the connection between Node1 and Node2 but not between Node2 and Node3.

It can be reproduced by the following test

```
// Create Node 1
        let mut conf_node1 = Conf::default();
        conf_node1.p2p = P2P::Yes;
        let node1 =
            BitcoinD::with_conf(bitcoind::downloaded_exe_path().unwrap(), &conf_node1).unwrap();

        // Create Node 2 connected Node 1
        let node1_connection = node1.p2p_connect().unwrap();
        let mut conf_node2 = Conf::default();
        conf_node2.p2p = node1_connection;
        let node2 =
            BitcoinD::with_conf(bitcoind::downloaded_exe_path().unwrap(), &conf_node2).unwrap();

        // Create Node 3 Connected To Node 2
        let node2_connection = node2.p2p_connect().unwrap();
        let mut conf_node3 = Conf::default();
        conf_node3.p2p = node2_connection;
        let node3 =
            BitcoinD::with_conf(bitcoind::downloaded_exe_path().unwrap(), &conf_node3).unwrap();

        // Get each nodes Peers
        let node1_peers = node1.client.get_peer_info().unwrap();
        let node2_peers = node2.client.get_peer_info().unwrap();
        let node3_peers = node3.client.get_peer_info().unwrap();

        // Peers found for Node 1 and 2, but not for Node 3
        assert_eq!(node1_peers.len(), 1);
        assert_eq!(node2_peers.len(), 1);
        assert_eq!(node3_peers.len(), 0);
````

### PR DESCRIPTION
**Pull request by RCasatta**
should fix #11

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -145,7 +145,8 @@ impl BitcoinD {
                 let p2p_socket = SocketAddrV4::new(LOCAL_IP, p2p_port);
                 let p2p_arg = format!("-port={}", p2p_port);
                 let connect = format!("-connect={}", other_node_url);
-                let args = vec![p2p_arg, connect];
+                let listen = "-listen=1".to_string(); // With connect specified listen is not defaulted to 1
```
Comment (LeoComandini) [inline src/lib.rs:148]: If we have always `listen=1` we can't setup network where some nodes are only connected to some other nodes, e.g.
`1 <--> 2 <--> 3` but `1 <-/-> 3`

So perhaps `listen` should be configurable by the user.
And also caller should be able to specify a set of node urls to connect to?

Comment (RCasatta) [reply src/lib.rs]: Added a boolean to the P2P::Connect variant that basically allow to specify if using listen=1 or listen=0

> And also caller should be able to specify a set of node urls to connect to?

Nodes with `listen=1` should spontaneously connect to each other by their own and one could use `addnode` to facilitate that so I think it's not super necessary specifying a set of nodes

File: src/lib.rs [hunk 2]
```diff
@@ -328,16 +332,48 @@ mod test {
...
+        let node3 = BitcoinD::with_conf(exe_path(), &conf_node3).unwrap();
+
+        // Get each nodes Peers
+        let node1_peers = node1.client.get_peer_info().unwrap();
+        let node2_peers = node2.client.get_peer_info().unwrap();
+        let node3_peers = node3.client.get_peer_info().unwrap();
+
+        // Peers found
+        assert!(node1_peers.len() >= 1);
+        assert!(node2_peers.len() >= 1);
```
Comment (LeoComandini) [inline src/lib.rs:372]: nit:
```suggestion
        assert_eq!(node1_peers.len(), 2);
        assert_eq!(node2_peers.len(), 2);
```

**Comment by LeoComandini** [comment]
utACK 84af7015b973a3b90f7797f226eb68cb0c6cc1b1 with nit

File: src/lib.rs (hunk 2 above)
Comment (RCasatta) [reply src/lib.rs:372]: it's `>= 1` because the node may or may not connect to each other in time

**Pull request merged**